### ISSUE TITLE
Make vcpkg usable from build scripts without user configuration

### ISSUE THREAD

**Issue by anordal** [microsoft/vcpkg#3380]
Use case:

A build script wants to consult vcpkg to install a package and then use it. It is uncool to encode user specific configuration in the build script.

Problems:

* It does not help that the user has ~~installed~~ git cloned vcpkg to some secret location.
* Even if vcpkg as a command is findable in PATH, it would be bad practice to assume it as a basis for the package installation path—I'm sure every linux distro packager would break that assumption! For the sake of argument, assume the wrapper in #3378.

Proposed solutions:

1. Add an option to query the package installation path or the vcpkg root. The wrapper script in #3378 actually does the latter, but for consistency, this is of course better done upstream.

or

2. Make it possible to override the package installation path. This should solve #1481 for free, and is more in line with how Meson subprojects work.

You might think option 2 would already work by overriding VCPKG_ROOT, but it doesn't:

```
> env VCPKG_ROOT=/tmp/somewhere vcpkg install utf8h
Error: invalid triplet: x64-linux
Available architecture triplets:
```

**Comment by wjk** [comment]
I was just thinking of this very issue. One solution I thought of was to write the absolute path to the directory containing `.vcpkg-root` to a file in a known location (say, `~/.vcpkg-root-location`) that can then be read in by MSBuild using `<ReadLinesFromFile />` or by CMake using `file(READ)`. Since this is under the user's home, it can be set for that user only, which is useful if vcpkg is in a per-user location, or if the user doesn't have local admin rights to the PC (so `vcpkg install integrate` won't work). I can kludge this on top of the existing infrastructure when using CMake, but having it available by default would be helpful.

I'm no expert on Linux packaging, so I offer no opinion on how this would affect machine-global installs.

**Comment by anordal** [comment]
Too bad I didn't see the fix, a9d989ec, which was made *the day before* I opened this issue … was I a git pull away?

My secret plan was [first-class Meson integration](https://github.com/mesonbuild/meson/issues/3500). Now that it's April again, Vcpkg could have had this two years ago!

Proof of concept:

```
vcroot=$(cat $HOME/.vcpkg/vcpkg.path.txt)
mkdir -p build/vc_subprojects
for i in triplets scripts ports; do
    ln -s $vcroot/$i build/vc_subprojects/$i
done
```
